test/integration/test_session.py

```diff
@@ -0,0 +1,77 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Integration tests for Session."""
+
+from qiskit.circuit.library import RealAmplitudes
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.test.reference_circuits import ReferenceCircuits
+
+from qiskit_ibm_runtime import (
+    Estimator,
+    EstimatorResult,
+    Session,
+    Sampler,
+    SamplerResult,
+)
+
+from ..decorators import run_integration_test
+from ..ibm_test_case import IBMIntegrationTestCase
+
+
+class TestIntegrationSession(IBMIntegrationTestCase):
+    """Integration tests for Session."""
+
+    @run_integration_test
+    def test_estimator_sampler(self, service):
+        """Test calling both estimator and sampler."""
+        options = {"backend": "ibmq_qasm_simulator"}
+
+        psi1 = RealAmplitudes(num_qubits=2, reps=2)
+        # pylint: disable=invalid-name
+        H1 = SparsePauliOp.from_list([("II", 1), ("IZ", 2), ("XI", 3)])
+        theta1 = [0, 1, 1, 2, 3, 5]
+
+        with Session(service) as session:
+            estimator = Estimator(session=session, options=options)
+            result = estimator.run(
+                circuits=[psi1], observables=[H1], parameter_values=[theta1], shots=100
+            ).result()
+            self.assertIsInstance(result, EstimatorResult)
+            self.assertEqual(len(result.values), 1)
+            self.assertEqual(len(result.metadata), 1)
+            self.assertEqual(result.metadata[0]["shots"], 100)
+
+            sampler = Sampler(session=session, options=options)
+            result = sampler.run(circuits=ReferenceCircuits.bell(), shots=200).result()
+            self.assertIsInstance(result, SamplerResult)
+            self.assertEqual(len(result.quasi_dists), 1)
+            self.assertEqual(len(result.metadata), 1)
+            self.assertEqual(result.metadata[0]["shots"], 200)
+            self.assertAlmostEqual(result.quasi_dists[0]["11"], 0.5, delta=0.05)
+            self.assertAlmostEqual(result.quasi_dists[0]["00"], 0.5, delta=0.05)
+
+            result = estimator.run(
+                circuits=[psi1], observables=[H1], parameter_values=[theta1], shots=300
+            ).result()
+            self.assertIsInstance(result, EstimatorResult)
+            self.assertEqual(len(result.values), 1)
+            self.assertEqual(len(result.metadata), 1)
+            self.assertEqual(result.metadata[0]["shots"], 300)
+
+            result = sampler.run(circuits=ReferenceCircuits.bell(), shots=400).result()
```

Comment:
```suggestion
            result = sampler.run(circuits=[ReferenceCircuits.bell()], shots=400).result()
```

would that be ok?

Reply:
sure both would work, but either way the test case still fails as we wait for the fix on the server side

Reply:
I mean, wrapping with array actually makes the merge happen correctly in the server. Because the sampler is NOT wrapping it as an array before it sends it. Should it?

Reply:
The documented interface says `circuits` can be "A single or list of QuantumCircuits". But we can wrap it if needed.